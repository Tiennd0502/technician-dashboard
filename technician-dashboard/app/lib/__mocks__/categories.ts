export const CATEGORIES = [
  {
    id: '01',
    label: 'Sofa Cleaning',
    image: 'https://i.ibb.co/QcSmTHW/category01.webp',
  },
  {
    id: '02',
    label: 'Bathroom',
    image: 'https://i.ibb.co/KDQzZ3s/category02.web',
  },
  {
    id: '03',
    label: 'Best Control',
    image: 'https://i.ibb.co/jwsPTTt/category03.webp',
  },
  {
    id: '04',
    label: 'View All',
    image: 'https://i.ibb.co/KxsM943/category04.webp',
  },
];
