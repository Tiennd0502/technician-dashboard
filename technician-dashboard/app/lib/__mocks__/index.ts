export * from './table';
export * from './categories';
export * from './services';
