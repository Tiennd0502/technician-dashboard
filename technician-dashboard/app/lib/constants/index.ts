export * from './common';
export * from './filter';
export * from './sidebar';
export * from './header';
export * from './rules';
export * from './messages';
export * from './routes';
export * from './status';
export * from './table';
export * from './categories';
export * from './services';
