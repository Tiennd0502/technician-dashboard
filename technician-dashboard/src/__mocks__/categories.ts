export const CATEGORIES = [
  {
    id: '01',
    label: 'Sofa Cleaning',
    image: 'https://i.ibb.co/dKnNFV1/category01.png',
  },
  {
    id: '02',
    label: 'Bathroom',
    image: 'https://i.ibb.co/Qdtfx8N/category02.png',
  },
  {
    id: '03',
    label: 'Best Control',
    image: 'https://i.ibb.co/dDLCwNf/category03.png',
  },
  {
    id: '04',
    label: 'View All',
    image: 'https://i.ibb.co/cDz3H5Q/category04.png',
  },
];
