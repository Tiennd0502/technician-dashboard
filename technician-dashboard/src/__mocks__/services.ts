export const SERVICE = {
  id: '01',
  name: 'Carpenter',
  price: 45,
  image: 'https://i.ibb.co/7WfPDMJ/service01.png',
};

export const SERVICES = [
  {
    id: '01',
    name: 'Carpenter',
    price: 45,
    image: 'https://i.ibb.co/7WfPDMJ/service01.png',
  },
  {
    id: '02',
    name: 'Ac Service & repair',
    price: 25,
    image: 'https://i.ibb.co/G069RKZ/service02.png',
  },
  {
    id: '03',
    name: 'Kitchecn Deep Cleaning',
    price: 30,
    image: 'https://i.ibb.co/nw3YSHS/service03.png',
  },
  {
    id: '04',
    name: 'Electrician',
    price: 50,
    image: 'https://i.ibb.co/Cvnw8GY/service04.png',
  },
];
