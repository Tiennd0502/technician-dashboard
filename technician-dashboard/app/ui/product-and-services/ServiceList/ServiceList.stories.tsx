import type { Meta, StoryObj } from '@storybook/react';

import { SERVICES } from '@/lib/constants';

import ServiceList from '.';

const meta: Meta<typeof ServiceList> = {
  title: 'Components/ServiceList',
  component: ServiceList,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof ServiceList>;

export const Default: Story = {
  args: { list: SERVICES },
};
