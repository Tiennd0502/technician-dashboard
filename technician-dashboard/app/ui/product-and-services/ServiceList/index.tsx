import { memo } from 'react';
import { Flex } from '@chakra-ui/react';

import { CURRENCY_CHARACTERS } from '@/lib/constants';
import { Service } from '@/lib/interfaces';
import { ServiceCard } from '..';

interface ServiceListProps {
  list: Service[];
}

const ServiceList = ({ list }: ServiceListProps) => (
  <Flex gap='5' data-testid='list-services'>
    {list.map((item) => (
      <ServiceCard key={item.id} service={item} currencyCharacter={CURRENCY_CHARACTERS.EURO} />
    ))}
  </Flex>
);

export default memo(ServiceList);
