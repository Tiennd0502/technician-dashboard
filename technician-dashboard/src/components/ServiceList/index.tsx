import { memo } from 'react';
import { Flex } from '@chakra-ui/react';

// Constants
import { CURRENCY_CHARACTERS } from '@/constants';

// Types
import { Service } from '@/interfaces';

// Components
import { ServiceCard } from '..';

interface ServiceListProps {
  list: Service[];
}

const ServiceList = ({ list }: ServiceListProps) => (
  <Flex gap='5' role='list'>
    {list.map((item) => (
      <ServiceCard key={item.id} service={item} currencyCharacter={CURRENCY_CHARACTERS.EURO} />
    ))}
  </Flex>
);

export default memo(ServiceList);
