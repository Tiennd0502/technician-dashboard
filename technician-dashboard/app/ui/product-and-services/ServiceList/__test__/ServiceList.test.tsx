import { render } from '@testing-library/react';

import { SERVICES } from '@/lib/__mocks__';
import ServiceList from '..';

describe('ServiceList test cases', () => {
  const props = {
    list: SERVICES,
  };

  test('should render correctly', () => {
    const { container } = render(<ServiceList {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('Should render enough elements', () => {
    const { getByTestId } = render(<ServiceList {...props} />);
    const serviceList = getByTestId('list-services');

    expect(serviceList.childNodes).toHaveLength(SERVICES.length);
  });

  test('Should render empty elements', () => {
    const { getByTestId } = render(<ServiceList list={[]} />);
    const serviceList = getByTestId('list-services');

    expect(serviceList.childNodes).toHaveLength(0);
  });
});
