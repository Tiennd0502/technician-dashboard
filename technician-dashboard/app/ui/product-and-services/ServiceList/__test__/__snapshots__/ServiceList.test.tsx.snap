// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceList test cases should render correctly 1`] = `
<div>
  <div
    class="css-1j52cd2"
    role="list"
  >
    <a
      class="chakra-link css-qwh07e"
      href="#"
    >
      <div
        class="css-1g4yje1"
        justify-content="flex-end"
      >
        <div
          class="css-17xejub"
        />
        <p
          class="chakra-text css-od0eia"
        >
          Carpenter
        </p>
        <p
          class="chakra-text css-18bew60"
        >
          €45
        </p>
      </div>
    </a>
    <a
      class="chakra-link css-17733pz"
      href="#"
    >
      <div
        class="css-1g4yje1"
        justify-content="flex-end"
      >
        <div
          class="css-17xejub"
        />
        <p
          class="chakra-text css-od0eia"
        >
          Ac Service & repair
        </p>
        <p
          class="chakra-text css-18bew60"
        >
          €25
        </p>
      </div>
    </a>
    <a
      class="chakra-link css-tu7nyg"
      href="#"
    >
      <div
        class="css-1g4yje1"
        justify-content="flex-end"
      >
        <div
          class="css-17xejub"
        />
        <p
          class="chakra-text css-od0eia"
        >
          Kitchecn Deep Cleaning
        </p>
        <p
          class="chakra-text css-18bew60"
        >
          €30
        </p>
      </div>
    </a>
    <a
      class="chakra-link css-tjcrj0"
      href="#"
    >
      <div
        class="css-1g4yje1"
        justify-content="flex-end"
      >
        <div
          class="css-17xejub"
        />
        <p
          class="chakra-text css-od0eia"
        >
          Electrician
        </p>
        <p
          class="chakra-text css-18bew60"
        >
          €50
        </p>
      </div>
    </a>
  </div>
</div>
`;
